apps/client/app/goods-receipts/[id]/page.tsx

```diff
@@ -0,0 +1,29 @@
+// app/goods-receipts/[id]/page.tsx
+import GoodsReceiptForm from '../../components/GoodsReceiptForm';
+import { fetchParentItems, fetchPurchaseOrders } from '../../utils/api';
+import { GoodsReceipt } from '../../interfaces';
+
+const fetchGoodsReceipt = async (id: string): Promise<GoodsReceipt> => {
+  const res = await fetch(`http://localhost:3100/api/goods-receipts/${id}`, { cache: 'no-cache' });
+  if (!res.ok) {
+    throw new Error('Failed to fetch goods receipt');
+  }
+  return res.json();
+};
+
+interface EditGoodsReceiptPageProps {
+  params: { id: string };
+}
+
+export default async function EditGoodsReceiptPage({ params }: EditGoodsReceiptPageProps) {
+  const goodsReceipt = await fetchGoodsReceipt(params.id);
+  const items = await fetchParentItems();
+  const purchaseOrders = await fetchPurchaseOrders();
```

Comment:
These 2 calls can be wrapped in a promise.all to make the page load faster

Reply:
Good suggestion. Done!